src/main/java/edu/usf/sas/pal/muser/ProcessorMain.java

```diff
@@ -1,67 +1,90 @@
                 return;
             }
+
             if (cmd.hasOption(ProgramOptions.USER_ID)) {
                 programOptions.setUserId(cmd.getOptionValue(ProgramOptions.USER_ID));
             }
+
             if (cmd.hasOption(ProgramOptions.START_DATE) && cmd.hasOption(ProgramOptions.END_DATE)) {
-                programOptions.setStartDate(cmd.getOptionValue(ProgramOptions.START_DATE));
-                programOptions.setEndDate(cmd.getOptionValue(ProgramOptions.END_DATE));
+                long dateStartMillis = StringUtils.validateStringDateAndParseToMillis(cmd.getOptionValue(ProgramOptions.START_DATE));
+                long dateEndMillis = StringUtils.validateStringDateAndParseToMillis(cmd.getOptionValue(ProgramOptions.END_DATE));
+
+                //Validate dates
+                if (dateStartMillis == 0 || dateEndMillis == 0) {
+                    System.err.println("Invalid start/end dates provided. \n" +
+                            "Please provide dates in using the format mm-dd-yyyy.");
+                    return;
+                }
+                programOptions.setStartDate(dateStartMillis);
+                programOptions.setEndDate(dateEndMillis);
+            } else if (cmd.hasOption(ProgramOptions.START_DATE)) {
+                System.err.println("startDate and endDate must be provided together. \n" +
+                        "startDate was provided but endDate was not provided.");
+                return;
+            } else if (cmd.hasOption(ProgramOptions.END_DATE)) {
+                System.err.println("startDate and endDate must be provided together. \n" +
+                        "endDate was provided but startDate was not provided.");
+                return;
             }
         } catch (ParseException e) {
             System.err.println("Invalid command line options");
         }
+
         System.out.println("Analysis started!");
         try {
             new MuserMusicDataAnalysisManager().processData();
         } catch (FirebaseFileNotInitializedException e) {
             System.err.println("Firebase file is not initialized properly.");
         }
+
         System.out.println("Analysis finished!");
     }
 
     private static Options createCommandLineOptions() {
         Options options = new Options();
         options.addOption(ProgramOptions.USER_ID, true, "Only run the analysis for specific user");
         options.addOption(ProgramOptions.KEY_FILE, true, "Admin key file of the Firebase account");
-        options.addOption(ProgramOptions.START_DATE, true, "Start date to collect filtered data based on date range and the format used to represent the start date is 'yyyy-MM-dd'");
-        options.addOption(ProgramOptions.END_DATE, true, "End date to collect filtered data based on date range and the format used to represent the end date is 'yyyy-MM-dd'");
+        options.addOption(ProgramOptions.START_DATE, true, "Start date (mm-dd-yyyy) to filter data collection based on a date range.");
```

Comment:
```suggestion
        options.addOption(ProgramOptions.START_DATE, true, "Start date (mm-dd-yyyy) to filter data collection based on a date range. Midnight UTC time is used for the provided date.");
```

Please also add similar note below for end date